src/components/Tooltip/Tooltip.tsx

```diff
@@ -178,7 +178,11 @@ const TooltipAnchor: FC<TooltipAnchorProps> = ({
     if (!isValidElement(children)) return;
 
     if (isTriggerInteractive) {
-      const props = context.getReferenceProps({ ref });
+      const props = context.getReferenceProps({
+        ref,
+        // Needed to fix https://github.com/element-hq/compound/issues/333
+        ...(typeof children.props === "object" ? children.props : {}),
```

Comment:
I'm unclear on why this works, for React 18. Why isn't it enough for `cloneElement` to merge these props? (Is the answer perhaps that `getReferenceProps` does some smarter merging of the event handlers it uses, internally?)